vllm/v1/worker/gpu_worker.py

```diff
@@ -168,6 +168,27 @@ def init_device(self):
         if self.device_config.device.type == "cuda":
             # This env var set by Ray causes exceptions with graph building.
             os.environ.pop("NCCL_ASYNC_ERROR_HANDLING", None)
+            if (
+                self.vllm_config.parallel_config.data_parallel_size > 1
+                and self.parallel_config.data_parallel_size_local > 0
+            ):
+                # Use local DP rank if available, otherwise use global DP rank.
+                data_parallel_local_rank: int = (
+                    self.parallel_config.data_parallel_rank_local
+                    if self.parallel_config.data_parallel_rank_local is not None
+                    else self.parallel_config.data_parallel_rank
+                ) % self.parallel_config.data_parallel_size_local
```

Comment:
suggest to simplify ... and I don't think we need the modulo
```suggestion
                dp_rank_local = self.parallel_config.data_parallel_rank_local
                if dp_rank_local is None:
                    dp_rank_local = self.parallel_config.data_parallel_rank
```